pkg/frontend/adminactions/kubeactions.go

```diff
@@ -72,6 +76,25 @@ func NewKubeActions(log *logrus.Entry, env env.Interface, oc *api.OpenShiftClust
 	}, nil
 }
 
+func (k *kubeActions) KubeGetPodLogs(ctx context.Context, namespace string, name string, containerName string, kubeInterface kubernetes.Interface) ([]byte, error) {
+	opts := corev1.PodLogOptions{Container: containerName}
+
+	if kubeInterface == nil {
+		err := api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "kubeclient's Clientset is empty in KubeActions")
+		k.log.Error(err)
+		return nil, err
+	}
+
+	r := kubeInterface.CoreV1().Pods(namespace).GetLogs(name, &opts)
+
+	result, err := r.Do(ctx).Raw()
+	if err != nil {
+		k.log.Error(err)
+		return nil, err
+	}
+	return result, nil
```

Comment:
I think we can simplify a bit:

```suggestion
	return r.Do(ctx).Raw()
```

Reply:
Yes .. this can be done but as we discussed this code won't be there as I will be moving the kubeInterface tot he kubeActions struct .. but yaa .. I will keep this in mind next time I write something like this ..